compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -281,10 +313,32 @@ public void visit(BLangFunction funcNode) {
             addToFunctionMap(funcNode, funcEnv, position, receiver.symbol, receiver.getBType());
         }
 
+        if (funcNode.flagSet.contains(Flag.LAMBDA)) {
+            rewriteParamsOfFunction(funcNode, funcEnv);
+        }
         funcNode.body = rewrite(funcNode.body, funcEnv);
         result = funcNode;
     }
 
+    public void rewriteParamsOfFunction(BLangFunction funcNode, SymbolEnv funcEnv) {
+        for (BLangSimpleVariable bLangSimpleVariable : funcNode.requiredParams) {
+            bLangSimpleVariable.typeNode = rewrite(bLangSimpleVariable.typeNode, funcEnv);
+            if (!Symbols.isFlagOn(bLangSimpleVariable.symbol.flags, Flags.DEFAULTABLE_PARAM)) {
+                continue;
+            }
+            Map<String, BLangLambdaFunction> defaultValues =
+                                                ((BInvokableTypeSymbol) funcNode.getBType().tsymbol).defaultValues;
```

Comment:
Can be extracted out to a variable outside the loop?

Reply:
because the type symbol can become null. due to that I kept within the for a loop.

Reply:
Removed with new changes.